apps/src/netsim/netsimNodeFactory.js

```diff
@@ -0,0 +1,41 @@
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+
+ maxlen: 90,
+ maxparams: 3,
+ maxstatements: 200
+ */
+'use strict';
+
+var netsimConstants = require('./netsimConstants');
+var NetSimClientNode = require('./NetSimClientNode');
+var NetSimRouterNode = require('./NetSimRouterNode');
+
+var NodeType = netsimConstants.NodeType;
+
+var netsimNodeFactory = module.exports;
+
+/**
+ * Given a set of rows from the node table on a shard, gives back a set of node
+ * controllers (of appropriate types).
+ * @param {!NetSimShard} shard
+ * @param {!Array.<Object>} rows
+ * @throws when a row doesn't have a mappable node type.
+ * @return {Array.<NetSimNode>} nodes for the rows
+ */
+netsimNodeFactory.nodesFromRows = function (shard, rows) {
+  return rows
+      .map(function (row) {
```

Comment:
I would just leave .map on the same line as rows here.

I also usually try to avoid having both foo and foos as variable names, since it makes it easy to typo and mix the two up.